### PR TITLE
Fix sgcheck for s390x and aarch64

### PR DESCRIPTION
**Pull request by grisu48**
This PR fixes the valgrind test for s390x and aarch64

On s390x and aarch64 sgcheck is not available, so we perform the `sgcheck` test only on `x86_64`.
This commit adds a check for `x86_64` and skips the test if not on that platform.

- Related ticket: N/A
- Needles: N/A
- Verification runs: 
[SLE15-SP1 s390x](https://openqa.suse.de/t4091245) | [SLE15 s390x](https://openqa.suse.de/t4091246)
[SLE15-SP1 x86_64](https://openqa.suse.de/t4084249) | [SLE15 x86_64](https://openqa.suse.de/t4084250) | [SLE12-SP5 x86_64](https://openqa.suse.de/t4084251) | [SLE12-SP4 x86_64](https://openqa.suse.de/t4084252)

### CODE REVIEW

**Comment by grisu48** [comment]
After the `sgcheck` also fails on aarch64 (See [poo](https://progress.opensuse.org/issues/65151#change-290013)) I changed the script, so that `sgcheck` is only performed on `x86_64` architectures

**Comment by okurz** [comment]
Please start the git commit message subject line with a capital letter and also mention aarch64 as you also target the fix for it

**Comment by grisu48** [comment]
I've updated the commit message accordingly. Didn't trigger new test runs, as nothing else was changed.

File: data/valgrind/valgrind-test.sh [hunk 1]
```diff
@@ -22,6 +22,15 @@ function GREP {
     fi
 }
 
+# Check for x86_64 architecture
+# Place this before set -e as it fails on other platforms
+x86_64=0
+uname -i > uname.txt
+grep -i x86_64 uname.txt >/dev/null && x86_64=1
+if [[ $x86_64 == 1 ]]; then
```
Comment (dzedro) [inline data/valgrind/valgrind-test.sh:30]: you could just run simply `if [ $(uname -m) == "x86_64" ] ...`

**Comment by okurz** [comment]
Sorry to still bring this up: Please see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages. Especially the subject line should be written in imperative mood, describing what the commit should achieve rather than what you did. At the very least the git commit message subject line should start with "Fix…" rather than "Fixes…"

File: data/valgrind/valgrind-test.sh (hunk 1 above)
Comment (grisu48) [reply data/valgrind/valgrind-test.sh]: I updated the test script accordingly and updated the test runs to run with the new version

**Comment by grisu48** [comment]
> Sorry to still bring this up: Please see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages. Especially the subject line should be written in imperative mood, describing what the commit should achieve rather than what you did. At the very least the git commit message subject line should start with "Fix…" rather than "Fixes…"

Fixed for the message subject line in the commit and in the PR.

File: data/valgrind/valgrind-test.sh [hunk 2]
```diff
@@ -109,5 +121,5 @@ GREP 'mem_heap_B=' massif.out
 GREP 'mem_heap_extra_B=' massif.out
 GREP 'heap_tree=' massif.out
 
-rm -f output_{0..8}.txt callgrind.out cachegrind.out massif.out valgrind-test
+rm -f output_{0..8}.txt callgrind.out cachegrind.out massif.out valgrind-test uname.txt
```
Comment (dzedro) [inline data/valgrind/valgrind-test.sh:124]: uname.txt not created anymore

**Pull request merged**